plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -443,10 +445,22 @@ private static long getSnapshotIdFromVersion(ConnectorSession session, Table tab
 
     private static long getTargetSnapshotIdFromVersion(Table table, ConnectorTableVersion version, io.trino.spi.type.Type versionType)
     {
-        if (versionType != BIGINT) {
+        long snapshotId;
+        if (versionType == BIGINT) {
+            snapshotId = (long) version.getVersion();
+        }
+        else if (versionType instanceof VarcharType) {
+            String refName = ((Slice) version.getVersion()).toStringUtf8();
+            SnapshotRef ref = table.refs().get(refName);
+            if (ref == null) {
+                throw new TrinoException(INVALID_ARGUMENTS, "Cannot find snapshot with reference name: " + refName);
+            }
+            snapshotId = ref.snapshotId();
```

Comment:
From this implementation, it seems that there is no distinction between branch and tag queries in trino, and both are used to query a fixed snapshot ?